py/torch_tensorrt/dynamo/_compiler.py

```diff
@@ -913,26 +945,61 @@ def contains_metadata(gm: torch.fx.GraphModule) -> bool:
         dryrun_tracker.tensorrt_graph_count += 1
         dryrun_tracker.per_subgraph_data.append(subgraph_data)
 
-        # Create TRT engines from submodule
+        # Create TRT engines / compiled models from submodule
+        # torch._logging.set_logs(inductor=logging.DEBUG)
         if not settings.dryrun:
-            trt_module = convert_module(
-                submodule,
-                submodule_inputs,
-                settings=settings,
-                name=name,
-                engine_cache=engine_cache,
-            )
+            if use_hierarchical_partitioner:
+                # compile submodule with pytorch inductor
+                if "_run_on_acc_inductor" in name:
+                    sub_inputs = []
+                    for input in submodule_inputs:
+                        sub_input = (
+                            torch.randn(input.shape)
+                            .to(dtype.to(input.dtype, t=torch.dtype))
+                            .cuda()
+                        )
+                        sub_inputs.append(sub_input)
+
+                    compiled_func = torch._inductor.compile(
+                        submodule,
+                        sub_inputs,
+                    )
```

Comment:
`torch.dynamo.mark_dynamic` API  is used to set dynamic shapes for torch.compile workflow. Reference: https://docs.pytorch.org/TensorRT/user_guide/dynamic_shapes.html. So you can use the construct_submodule_inputs() API to give you dynamic inputs (if that's the case) and set them to the inductor segment accordingly.